migrations/2023-06-23-172550_expire_password_reset_tokens/up.sql

```diff
@@ -0,0 +1,2 @@
+alter table password_reset_request add column expires_at timestamp not null;
+create index idx_password_reset_request_token_encrypted on password_reset_request using hash (token_encrypted);
```

Comment:
using hash? Why not just a regular index?

Reply:
Another thing: why not just generate the UUID in postgres, instead of rust?

Reply:
> using hash? Why not just a regular index?

Btree index seems a waste since ordering by hashed tokens doesn't seem useful in any way, hash index is more performant for direct access.

> Another thing: why not just generate the UUID in postgres, instead of rust?

A couple of reasons, one is so that it can be hashed before inserting, and another is that it's [not clear](https://security.stackexchange.com/a/93905) if postgres's UUID4 generation is guaranteed to be cryptographically secure.